lupa/_lupa.pyx

```diff
@@ -276,16 +299,55 @@ cdef class LuaRuntime:
                 raise ValueError("attribute_filter and attribute_handlers are mutually exclusive")
             self._attribute_getter, self._attribute_setter = getter, setter
 
+        lua.lua_atpanic(L, &_lua_panic)
         lua.luaL_openlibs(L)
         self.init_python_lib(register_eval, register_builtins)
-        lua.lua_atpanic(L, <lua.lua_CFunction>1)
 
         self.set_overflow_handler(overflow_handler)
 
+        # lupa init done, set real limit
+        if max_memory is not None:
+            self._memory_status.base_usage = self._memory_status.used
+            if max_memory > 0:
+                self._memory_status.limit =  self._memory_status.base_usage + <size_t>max_memory
+
     def __dealloc__(self):
         if self._state is not NULL:
             lua.lua_close(self._state)
             self._state = NULL
+        if self._memory_status is not NULL:
+            free(self._memory_status)
+            self._memory_status = NULL
+
+    def get_max_memory(self, count_base=False):
```

Comment:
To me, `count_base` reads like `base of count`, not `count the base`. Maybe `total` would be a better name? I think that's what it is, the total memory including the setup size of the runtime.